future-apps/payroll/contracts/Payroll.sol

```diff
@@ -112,119 +147,158 @@ contract Payroll is AragonApp { //, IForwarder { // makes coverage crash (remove
-     * @param initialYearlyDenominationSalary Employee's salary
+     * @param initialDenominationSalary Employee's salary, per second in denomination Token
      * @param name Employee's name
      */
     function addEmployeeWithName(
         address accountAddress,
-        uint256 initialYearlyDenominationSalary,
+        uint256 initialDenominationSalary,
         string name
     )
         external
-        authP(ADD_EMPLOYEE_ROLE, arr(initialYearlyDenominationSalary, getTimestamp()))
+        authP(ADD_EMPLOYEE_ROLE, arr(accountAddress, initialDenominationSalary, getTimestamp64()))
     {
         _addEmployee(
             accountAddress,
-            initialYearlyDenominationSalary,
+            initialDenominationSalary,
             name,
-            getTimestamp()
+            getTimestamp64()
         );
     }
 
     /**
      * @dev Add employee to Payroll
-     * @notice Creates employee, adds it to mappings, initializes values.
-               Updates also global Payroll salary sum.
+     * @notice Add employee `name` with address `accountAddress` to Payroll with a salary of `initialDenominationSalary` per second, starting on `startDate`.
      * @param accountAddress Employee's address to receive Payroll
-     * @param initialYearlyDenominationSalary Employee's salary
+     * @param initialDenominationSalary Employee's salary, per second in denomintation Token
      * @param name Employee's name
      * @param startDate It will actually set initial lastPayroll value
      */
     function addEmployeeWithNameAndStartDate(
         address accountAddress,
-        uint256 initialYearlyDenominationSalary,
+        uint256 initialDenominationSalary,
         string name,
-        uint256 startDate
+        uint64 startDate
     )
         external
-        authP(ADD_EMPLOYEE_ROLE, arr(initialYearlyDenominationSalary, startDate))
+        authP(ADD_EMPLOYEE_ROLE, arr(accountAddress, initialDenominationSalary, startDate))
     {
         _addEmployee(
             accountAddress,
-            initialYearlyDenominationSalary,
+            initialDenominationSalary,
             name,
             startDate
         );
     }
 
     /**
      * @dev Set employee's annual salary
-     * @notice Updates also global Payroll salary sum
+     * @notice Set employee #`employeeId` annual salary to `denominationSalary` per second.
      * @param employeeId Employee's identifier
-     * @param yearlyDenominationSalary Employee's new salary
+     * @param denominationSalary Employee's new salary, per second in denomintation Token
      */
     function setEmployeeSalary(
         uint128 employeeId,
-        uint256 yearlyDenominationSalary
+        uint256 denominationSalary
     )
+        employeeExists(employeeId)
         external
-        authP(ADD_EMPLOYEE_ROLE, arr(yearlyDenominationSalary, 0))
+        authP(SET_EMPLOYEE_SALARY_ROLE, arr(uint256(employeeId), denominationSalary))
```

Comment:
I don't think we need to cast `employeeId` here, but in case we do, I would also cast `uint64` dates in `arr` around.